### ISSUE TITLE
Xiaomi HyperOs: accessibility service not working

### ISSUE THREAD

**Issue by mxr576** [d4rken-org/sdmaid-se#936]
After the Xiaomi HyperOs update the accessibility service stopped working. Even after a fresh reboot, if I start the app immediately, it is just not working.

![Screenshot_2024-01-11-08-18-15-334_com android settings](https://github.com/d4rken-org/sdmaid/assets/1755573/e2f25375-170d-48e3-8b62-476d243d2ac7)

**Comment by mxr576** [comment]
This is a copy-paste of https://github.com/d4rken-org/sdmaid/issues/6197

**Comment by mxr576** [comment]
I guess the relevant part of the debug log is:

```

boundsInScreen: Rect(0, 0 - 1220, 2712); boundsInWindow: Rect(0, 0 - 1220, 2712); packageName: com.miui.securitycenter; className: android.widget.FrameLayout; text: null; error: null; maxTextLength: -1; stateDescription: null; contentDescription: null; tooltipText: null; containerTitle: null; viewIdResName: null; uniqueId: null; checkable: false; checked: false; focusable: false; focused: false; selected: false; clickable: false; longClickable: false; contextClickable: false; enabled: true; password: false; scrollable: false; importantForAccessibility: true; visible: true; actions: [AccessibilityAction: ACTION_SELECT - null, AccessibilityAction: ACTION_CLEAR_SELECTION - null, AccessibilityAction: ACTION_ACCESSIBILITY_FOCUS - null, AccessibilityAction: ACTION_SHOW_ON_SCREEN - null]; isTextSelectable: false (spec=Spec(parent=SDMSE:AppCleaner:Automation:AOSP:Specs, label=Find & click 'Storage' (targets=[Storage & cache])))
2024-01-16T20:12:25.610Z  V/SDMSE:Automation:Service: Providing window root: android.view.accessibility.AccessibilityNodeInfo@80006d12; boundsInParent: Rect(0, 0 - 1220, 2712); boundsInScreen: Rect(0, 0 - 1220, 2712); boundsInWindow: Rect(0, 0 - 1220, 2712); packageName: com.miui.securitycenter; className: android.widget.FrameLayout; text: null; error: null; maxTextLength: -1; stateDescription: null; contentDescription: null; tooltipText: null; containerTitle: null; viewIdResName: null; uniqueId: null; checkable: false; checked: false; focusable: false; focused: false; selected: false; clickable: false; longClickable: false; contextClickable: false; enabled: true; password: false; scrollable: false; importantForAccessibility: true; visible: true; actions: [AccessibilityAction: ACTION_SELECT - null, AccessibilityAction: ACTION_CLEAR_SELECTION - null, AccessibilityAction: ACTION_ACCESSIBILITY_FOCUS - null, AccessibilityAction: ACTION_SHOW_ON_SCREEN - null]; isTextSelectable: false
2024-01-16T20:12:25.610Z  V/SDMSE:Automation:Crawler: Looking for viable window root, current nodes:
2024-01-16T20:12:25.611Z  V/SDMSE:Automation:Crawler: crawl():0: className=android.widget.FrameLayout, text='null', isClickable=false, isEnabled=true, viewIdResourceName=null, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.611Z  V/SDMSE:Automation:Crawler: crawl():-1: className=android.view.ViewGroup, text='null', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/action_bar, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.612Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.FrameLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/up, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.612Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.ImageView, text='null', isClickable=true, isEnabled=true, viewIdResourceName=null, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.612Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.TextView, text='App info', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/action_bar_title_expand, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.613Z  V/SDMSE:Automation:Crawler: crawl():-1: className=android.widget.ScrollView, text='null', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_scrollview, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.615Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.TextView, text='Firefox', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/app_manager_details_applabel, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.615Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.TextView, text='Version: 121.1.0', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/app_manager_details_appversion, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.615Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_storage_view, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.616Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Storage', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.617Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='892 MB', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_summary, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.617Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_traffic_view, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.617Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Data usage', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.617Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='67.1MB', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_summary, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.617Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_power_view, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.618Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Battery', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.618Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='29.8%', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_summary, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.618Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.TextView, text='Permissions', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/tv_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.618Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_detail_hibernations_switch, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.619Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Pause app activity if unused', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/tv_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.619Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Remove permissions, delete temporary files, and stop notifications', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/tv_summary, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.619Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.CheckBox, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_switch, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.619Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_global_perm, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.619Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='App permissions', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/tv_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.620Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Manage permissions related to location, storage, phone, messages, and contacts.', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/tv_summary, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.620Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_detail_perm, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.620Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Other permissions', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.620Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='null', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_summary, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.620Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_detail_notify, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.621Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Notifications', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.621Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Yes', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_summary, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.621Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_detail_net, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.621Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Connection method', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.621Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Wi-Fi, Mobile data', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_summary, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.621Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_detail_keeper, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.621Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Battery saver', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.621Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='No restrictions', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_summary, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.621Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.TextView, text='Advanced settings', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/tv_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.621Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_full_screen, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.622Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Fullscreen mode', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/tv_title, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.622Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.TextView, text='Allow selected apps to work in fullscreen mode', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/tv_summary, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.622Z  V/SDMSE:Automation:Crawler: crawl():---3: className=android.widget.CheckBox, text='null', isClickable=true, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/am_switch, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.622Z  V/SDMSE:Automation:Crawler: crawl():-1: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=null, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.622Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.TextView, text='Force stop', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/action_menu_item_child_text, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.623Z  V/SDMSE:Automation:Crawler: crawl():-1: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=null, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.623Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.TextView, text='Uninstall', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/action_menu_item_child_text, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.623Z  V/SDMSE:Automation:Crawler: crawl():-1: className=android.widget.LinearLayout, text='null', isClickable=true, isEnabled=true, viewIdResourceName=null, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.623Z  V/SDMSE:Automation:Crawler: crawl():--2: className=android.widget.TextView, text='Clear data', isClickable=false, isEnabled=true, viewIdResourceName=com.miui.securitycenter:id/action_menu_item_child_text, pkgName=com.miui.securitycenter
2024-01-16T20:12:25.623Z  D/SDMSE:Automation:Crawler: Not a viable root node: android.view.accessibility.AccessibilityNodeInfo@80006d12; boundsInParent: Rect(0, 0 - 1220, 2712); boundsInScreen: Rect(0, 0 - 1220, 2712); boundsInWindow: Rect(0, 0 - 1220, 2712); packageName: com.miui.securitycenter; className: android.widget.FrameLayout; text: null; error: null; maxTextLength: -1; stateDescription: null; contentDescription: null; tooltipText: null; containerTitle: null; viewIdResName: null; uniqueId: null; checkable: false; checked: false; focusable: false; focused: false; selected: false; clickable: false; longClickable: false; contextClickable: false; enabled: true; password: false; scrollable: false; importantForAccessibility: true; visible: true; actions: [AccessibilityAction: ACTION_SELECT - null, AccessibilityAction: ACTION_CLEAR_SELECTION - null, AccessibilityAction: ACTION_ACCESSIBILITY_FOCUS - null, AccessibilityAction: ACTION_SHOW_ON_SCREEN - null]; isTextSelectable: false (spec=Spec(parent=SDMSE:AppCleaner:Automation:AOSP:Specs, label=Find & click 'Storage' (targets=[Storage & cache])))
2024-01-16T20:12:25.721Z  D/SDMSE:Automation:ScreenState: unregisterReceiver(eu.darken.sdmse.automation.core.ScreenState$state$1$receiver$1@f42d36a)
2024-01-16T20:12:25.721Z  W/SDMSE:Automation:Crawler: crawl(): Attempt 4 failed on Spec(parent=SDMSE:AppCleaner:Automation:AOSP:Specs, label=Find & click 'Storage' (targets=[Storage & cache])):
kotlinx.coroutines.JobCancellationException: DeferredCoroutine was cancelled; job=DeferredCoroutine{Cancelling}@6f2e95d

```

**Comment by d4rken** [comment]
The ACS screenshot is from SD Maid 1, but that's just a mistake right?
Did you try v0.18.3-beta0 or later? already worked on adding some improved HyperOS support.

In your log it uses AOSP matching still: `parent=SDMSE:AppCleaner:Automation:AOSP:Specs`

I guess we can introduce new specs just for HyperOS, so if the API level is Android 14+ (is HyperOS also available on Android 13?) and some other criteria to differentiate it from AOSP :thinking: 

What is the full finger print of your ROM type? It's in the beginning of SD Maids log

**Comment by mxr576** [comment]
Yes, the screenshot is from the older version but the issue was the same. This should answer the SDMaid version questions as well if I am not mistaken:

```
Logfile: /storage/emulated/0/Android/data/eu.darken.sdmse/cache/debug/logs/eu.darken.sdmse_logfile_1705435899967.log
2024-01-16T20:11:39.998Z  I/SDMSE:Logging:Default: Was installed FileLogger(file=/storage/emulated/0/Android/data/eu.darken.sdmse/cache/debug/logs/eu.darken.sdmse_logfile_1705435899967.log)
2024-01-16T20:11:39.998Z  I/SDMSE:Debug:Log:eu.darken.sdmse.common.debug.recording.core.Recorder: Now logging to file!
2024-01-16T20:11:40.003Z  I/SDMSE:Debug:Log:Recorder:Module: APILEVEL: 34
2024-01-16T20:11:40.003Z  I/SDMSE:Debug:Log:Recorder:Module: Build.FINGERPRINT: Xiaomi/plato_eea/plato:14/UP1A.230905.011/V816.0.5.0.ULQEUXM:user/release-keys
2024-01-16T20:11:40.003Z  I/SDMSE:Debug:Log:Recorder:Module: Build.MANUFACTOR: Xiaomi
2024-01-16T20:11:40.003Z  I/SDMSE:Debug:Log:Recorder:Module: Build.BRAND: Xiaomi
2024-01-16T20:11:40.003Z  I/SDMSE:Debug:Log:Recorder:Module: Build.PRODUCT: plato_eea
2024-01-16T20:11:40.003Z  I/SDMSE:Debug:Log:Recorder:Module: App: eu.darken.sdmse - 0.18.2-beta0 (1802000) 
2024-01-16T20:11:40.003Z  I/SDMSE:Debug:Log:Recorder:Module: Build: GPLAY-BETA
2024-01-16T20:11:40.007Z  D/SDMSE:InstallID: New install ID created: 6ddcd0e9-1275-42ad-8703-872a41c6c062
2024-01-16T20:11:40.008Z  I/SDMSE:Debug:Log:Recorder:Module: Install ID: 6ddcd0e9-1275-42ad-8703-872a41c6c062
2024-01-16T20:11:40.008Z  I/SDMSE:Debug:Log:Recorder:Module: App locales: [en_US]
2024-01-16T20:11:40.008Z  I/SDMSE:Debug:Log:Recorder:Module: Data areas: (2)
```

**Comment by mxr576** [comment]
It looks like the issue is fixed on 0.18.beta0. Maybe since 0.18.3
